src/blockchain/adapters.ts

```diff
@@ -1,20 +1,27 @@
-import AppConfig from '../../config';
-
 import { Adapter as BitcoinAdapter } from '@jelly-swap/bitcoin';
-import { Adapter as EthereumAdapter } from '@jelly-swap/ethereum';
+import { Adapter as EthereumAdapter, Adapter } from '@jelly-swap/ethereum';
 import { Adapter as AeternityAdapter } from '@jelly-swap/aeternity';
 import { Adapter as Erc20Adapter } from '@jelly-swap/erc20';
-import Config from './config';
 
-const AllAdapters = {
-    BTC: AppConfig.NETWORKS.BTC ? new BitcoinAdapter(Config.BTC) : null,
-    ETH: AppConfig.NETWORKS.ETH ? new EthereumAdapter(Config.ETH) : null,
-    AE: AppConfig.NETWORKS.AE ? new AeternityAdapter(Config.AE) : null,
-    DAI: AppConfig.NETWORKS.DAI ? new Erc20Adapter('DAI', Config.DAI) : null,
-    USDC: AppConfig.NETWORKS.USDC ? new Erc20Adapter('USDC', Config.USDC) : null,
-    WBTC: AppConfig.NETWORKS.WBTC ? new Erc20Adapter('WBTC', Config.WBTC) : null,
-};
+import getConfig from './config';
+
+let Adapters: any;
 
-const Adapters = Object.entries(AllAdapters).reduce((a, [k, v]) => (v === null ? a : { ...a, [k]: v }), {});
+export default () => {
+    if (!Adapters) {
+        const Config = getConfig();
 
-export default Adapters;
+        const AllAdapters = {
+            ETH: Config.ETH && new EthereumAdapter(Config.ETH as any),
+            BTC: Config.BTC && new BitcoinAdapter(Config.BTC as any),
+            AE: Config.AE && new AeternityAdapter(Config.AE as any),
+            DAI: Config.DAI && new Erc20Adapter('DAI', Config.DAI),
```

Comment:
Should be 'Config.DAI as any' to be consistent?

Reply:
The reason for `as any` is because of typescript not being able to detect that the config is first checked and then used.